packages/vx-tooltip/src/enhancers/withTooltip.tsx

```diff
@@ -33,61 +20,11 @@ export default function withTooltip<BaseComponentProps = {}, TooltipData = {}>(
   },
   renderContainer: RenderTooltipContainer = (children, props) => <div {...props}>{children}</div>,
 ) {
-  return class WrappedComponent extends React.PureComponent<
-    BaseComponentProps,
-    WithTooltipState<TooltipData>
-  > {
-    state = {
-      tooltipOpen: false,
-      tooltipLeft: undefined,
-      tooltipTop: undefined,
-      tooltipData: undefined,
-    };
+  const WrappedComponent: FunctionComponent<BaseComponentProps> = props => {
+    const tooltipProps = useTooltip<TooltipData>();
 
-    updateTooltip = ({
-      tooltipOpen,
-      tooltipLeft,
-      tooltipTop,
-      tooltipData,
-    }: UpdateTooltipArgs<TooltipData>) => {
-      this.setState(prevState => ({
-        ...prevState,
-        tooltipOpen,
-        tooltipLeft,
-        tooltipTop,
-        tooltipData,
-      }));
-    };
-
-    showTooltip = ({ tooltipLeft, tooltipTop, tooltipData }: ShowTooltipArgs<TooltipData>) => {
-      this.updateTooltip({
-        tooltipOpen: true,
-        tooltipLeft,
-        tooltipTop,
-        tooltipData,
-      });
-    };
-
-    hideTooltip = () => {
-      this.updateTooltip({
-        tooltipOpen: false,
-        tooltipLeft: undefined,
-        tooltipTop: undefined,
-        tooltipData: undefined,
-      });
-    };
-
-    render() {
-      return renderContainer(
-        <BaseComponent
-          updateTooltip={this.updateTooltip}
-          showTooltip={this.showTooltip}
-          hideTooltip={this.hideTooltip}
-          {...this.state}
-          {...this.props}
-        />,
-        containerProps,
-      );
-    }
+    return renderContainer(<BaseComponent {...props} {...tooltipProps} />, containerProps);
```

Comment:
just noting that previously we spread the `WrappedComponent` `props` _after_ `tooltipProps`, so this might result in a change in behavior if a consumer did pass an overlapping `prop`. I'm not really sure this is a big concern, but would prevent doing things like overriding `tooltipOpen` 🤷‍♂  any thoughts on this?

Reply:
Ah, good catch!  I agree it's not super likely that any consumers are relying on that behavior now, but in the absence of a reason to change it I think it makes sense to stay consistent with the existing behavior.  Will update shortly.

Reply:
sounds good, thanks!